src/client/Microsoft.Identity.Client/ApiConfig/Executors/ManagedIdentityExecutor.cs

```diff
@@ -35,19 +38,48 @@ public ManagedIdentityExecutor(IServiceBundle serviceBundle, ManagedIdentityAppl
         {
             var requestContext = CreateRequestContextAndLogVersionInfo(commonParameters.CorrelationId, cancellationToken);
 
+            if (_managedIdentityApplication.KeyMaterialManager.CryptoKeyType != CryptoKeyType.None)
+            {
+                // managed identity resource 
+                string miResource = managedIdentityParameters.Resource;
+
+                // Check if the input ends with "/.default"
+                bool endsWithDefault = miResource.EndsWith("/.default", StringComparison.OrdinalIgnoreCase);
+
+                // Add "/.default" only if it doesn't end with "/.default" and doesn't contain "/.somethingelse"
+                if (!endsWithDefault)
+                {
+                    // Add "/.default" to the scopes
+                    commonParameters.Scopes = new SortedSet<string>
+                    {
+                        managedIdentityParameters.Resource + "/.default"
+                    };
```

Comment:
You can add this as a method to check and add ./default suffix in class [ScopeHelper.cs](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/main/src/client/Microsoft.Identity.Client/Utils/ScopeHelper.cs).

Reply:
Is this a requirement for the SLC endpoint to have the ./default suffix?

Reply:
yes, since we are calling ests directly